File: frontend/de/unika/ipd/grgen/util/Annotations.java
/*
 * GrGen: graph rewrite generator tool -- release GrGen.NET 3.6
 * Copyright (C) 2003-2013 Universitaet Karlsruhe, Institut fuer Programmstrukturen und Datenorganisation, LS Goos; and free programmers
 * licensed under LGPL v3 (see LICENSE.txt included in the packaging of this file)
 * www.grgen.net
 */

/**
 * @author Sebastian Hack
 */

package de.unika.ipd.grgen.util;

import java.util.Set;

/**
 * A collection of annotations.
 */
public interface Annotations {

	boolean containsKey(String key);

	Object get(String key);

	boolean isInteger(String key);

	boolean isBoolean(String key);

	boolean isString(String key);

	boolean isFlagSet(String key);

	void put(String key, Object value);

	Set<String> keySet();
}

File: frontend/de/unika/ipd/grgen/util/Annotations.cs
/*
 * GrGen: graph rewrite generator tool -- release GrGen.NET 8.1
 * Copyright (C) 2003-2026 Universitaet Karlsruhe, Institut fuer Programmstrukturen und Datenorganisation, LS Goos; and free programmers
 * licensed under LGPL v3, some components/parts use different licenses (see LICENSE.txt included in the packaging of this file)
 * www.grgen.de / www.grgen.net
 */

/// <summary>
/// @author Sebastian Hack
/// </summary>

namespace de.unika.ipd.grgen.util
{

using System.Collections.Generic;

/// <summary>
/// A collection of annotations.
/// </summary>
public interface Annotations
{
	bool ContainsKey(string key);

	object Get(string key);

	bool IsInteger(string key);

	bool IsBoolean(string key);

	bool IsString(string key);

	bool IsFlagSet(string key);

	void Put(string key, object value);

	ISet<string> KeySet();
}

}
